common/src/main/java/io/netty/util/CharsetUtil.java

```diff
@@ -67,51 +68,113 @@
     public static Charset[] values() { return CHARSETS; }
 
     /**
-     * Returns a cached thread-local {@link CharsetEncoder} for the specified
-     * <tt>charset</tt>.
+     * @deprecated Use {@link #encoder(Charset)}.
      */
+    @Deprecated
     public static CharsetEncoder getEncoder(Charset charset) {
-        if (charset == null) {
-            throw new NullPointerException("charset");
-        }
+        return encoder(charset);
+    }
+
+    /**
+     * Returns a new {@link CharsetEncoder} for the <code>charset</code> with specified error actions.
+     *
+     * @param charset The specified charset
+     * @param malformedInputAction The encoder's action for malformed-input errors
+     * @param unmappableCharacterAction The encoder's action for unmappable-character errors
+     * @return The encoder for the specified <code>charset</code>
+     */
+    public static CharsetEncoder encoder(Charset charset, CodingErrorAction malformedInputAction,
+                                         CodingErrorAction unmappableCharacterAction) {
+        checkNotNull(charset, "charset");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "charset" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1192)